eth-contracts/contracts/CKBChain.sol

```diff
@@ -106,29 +322,47 @@ contract CKBChain is ICKBChain, ICKBSpv {
 
         // calc the transactionsRoot by [rawTransactionsRoot, witnessesRoot]
         bytes32 transactionsRoot = CKBCrypto.digest(abi.encodePacked(rawTxRoot, proofView.witnessesRoot()), 64);
-        require(transactionsRoot == canonicalTransactionRoots[blockHash], "proof not passed");
+        require(transactionsRoot == canonicalTransactionsRoots[blockHash], "proof not passed");
         return true;
     }
 
-    /// #GC
-    /// Remove hashes from the active chain that are at least as old as the given header number.
-    function gcActiveChain(uint64 blockNumber) internal {
+    /// Remove hashes from the Canonical chain that are at least as old as the given header number.
+    function _gcCanonicalChain(uint64 blockNumber) internal {
```

Comment:
better remove gc, because delete map value is the same as set map value to zero

Reply:
I'll evaluate it